File: src/main/java/org/processmining/models/descriptions/BasicBPMNGenerationDescription.java
package org.processmining.models.descriptions;

/**
 * Created by Ivan Shugurov on 30.12.2014.
 */
public class BasicBPMNGenerationDescription extends BaseGenerationDescription
{
    private boolean usingResources;

    public BasicBPMNGenerationDescription()
    {
    }

    public BasicBPMNGenerationDescription(boolean usingResources)
    {
        this.usingResources = usingResources;
    }

    @Override
    public boolean isUsingTime()
    {
        return false;
    }

    @Override
    public boolean isUsingResources()
    {
        return usingResources;
    }

    public void setUsingResources(boolean usingResources)
    {
        this.usingResources = usingResources;
    }

    @Override
    public boolean isUsingLifecycle()
    {
        return false;
    }

    @Override
    public boolean isRemovingEmptyTraces()
    {
        return true;
    }

    @Override
    public boolean isRemovingUnfinishedTraces()
    {
        return true;
    }
}

File: console/src/main/kotlin/org/processmining/models/descriptions/BasicBPMNGenerationDescription.kt
package org.processmining.models.descriptions

/**
 * Created by Ivan Shugurov on 30.12.2014.
 */
open class BasicBPMNGenerationDescription(
        override var isUsingResources: Boolean = false
) : BaseGenerationDescription() {
//    override var isUsingResources: Boolean = false
    
    override val isUsingTime: Boolean = false
    
    override val isUsingLifecycle: Boolean = false
    
    override val isRemovingEmptyTraces: Boolean = true
    
    override val isRemovingUnfinishedTraces: Boolean = true
}
